Some SocketOptions not supported on Windows

**Jetty version(s)**
10.0.x

**Java version/vendor** `(use: java -version)`
11.0.12

**OS type/version**
Windows 10

**Description**

The recent work on Issue #6372 is not compatible with Windows 10.

```
Caused by: java.lang.UnsupportedOperationException: 'SO_REUSEPORT' not supported
	at java.base/sun.nio.ch.ServerSocketChannelImpl.setOption(ServerSocketChannelImpl.java:151)
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:338)
```

--- a/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java
+++ b/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java
@@ -486,7 +486,7 @@ public class ClientConnector extends ContainerLifeCycle
         catch (Throwable x)
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("Could not configure {} to {} on {}", option, value, channel);
+                LOG.debug("Could not configure {} to {} on {}", option, value, channel, x);
         }
     }
 

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java
@@ -19,6 +19,7 @@ import java.net.InetSocketAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
 import java.net.SocketException;
+import java.net.SocketOption;
 import java.net.StandardSocketOptions;
 import java.nio.channels.Channel;
 import java.nio.channels.SelectableChannel;
@@ -332,10 +333,10 @@ public class ServerConnector extends AbstractNetworkConnector
         {
             InetSocketAddress bindAddress = getHost() == null ? new InetSocketAddress(getPort()) : new InetSocketAddress(getHost(), getPort());
             serverChannel = ServerSocketChannel.open();
+            setSocketOption(serverChannel, StandardSocketOptions.SO_REUSEADDR, getReuseAddress());
+            setSocketOption(serverChannel, StandardSocketOptions.SO_REUSEPORT, isReusePort());
             try
             {
-                serverChannel.setOption(StandardSocketOptions.SO_REUSEADDR, getReuseAddress());
-                serverChannel.setOption(StandardSocketOptions.SO_REUSEPORT, isReusePort());
                 serverChannel.bind(bindAddress, getAcceptQueueSize());
             }
             catch (Throwable e)
@@ -346,6 +347,18 @@ public class ServerConnector extends AbstractNetworkConnector
         }
 
         return serverChannel;
+    }
+
+    private <T> void setSocketOption(ServerSocketChannel channel, SocketOption<T> name, T value)
+    {
+        try
+        {
+            channel.setOption(name, value);
+        }
+        catch (Throwable t)
+        {
+            LOG.warn("Unable to set socket option {} to {}", name, value, t);
+        }
     }
 
     @Override

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java
@@ -349,15 +349,16 @@ public class ServerConnector extends AbstractNetworkConnector
         return serverChannel;
     }
 
-    private <T> void setSocketOption(ServerSocketChannel channel, SocketOption<T> name, T value)
+    private <T> void setSocketOption(ServerSocketChannel channel, SocketOption<T> option, T value)
     {
         try
         {
-            channel.setOption(name, value);
+            channel.setOption(option, value);
         }
-        catch (Throwable t)
+        catch (Throwable x)
         {
-            LOG.warn("Unable to set socket option {} to {}", name, value, t);
+            if (LOG.isDebugEnabled())
+                LOG.debug("Could not configure {} to {} on {}", option, value, channel, x);
         }
     }
 

